src/Analyzers/CSharp/Analyzers/RemoveUnreachableCode/RemoveUnreachableCodeHelpers.cs

```diff
@@ -12,15 +14,35 @@ internal static class RemoveUnreachableCodeHelpers
     {
         public static ImmutableArray<ImmutableArray<StatementSyntax>> GetSubsequentUnreachableSections(StatementSyntax firstUnreachableStatement)
         {
-            SyntaxList<StatementSyntax> siblingStatements;
+            StatementSyntax[] siblingStatements;
             switch (firstUnreachableStatement.Parent)
             {
                 case BlockSyntax block:
-                    siblingStatements = block.Statements;
+                    siblingStatements = block.Statements.ToArray();
                     break;
 
                 case SwitchSectionSyntax switchSection:
-                    siblingStatements = switchSection.Statements;
+                    siblingStatements = switchSection.Statements.ToArray();
+                    break;
+
+                case GlobalStatementSyntax globalStatement:
+                    if (globalStatement.Parent is not CompilationUnitSyntax compilationUnit)
+                    {
+                        return ImmutableArray<ImmutableArray<StatementSyntax>>.Empty;
+                    }
+
+                    var builder = ArrayBuilder<StatementSyntax>.GetInstance();
```

Comment:
```suggestion
                    using var _ = ArrayBuilder<StatementSyntax>.GetInstance(out var builder);
```

you'll likely need to surround with a block as well.